backend/open_webui/apps/openai/main.py

```diff
@@ -407,20 +405,25 @@ async def generate_chat_completion(
 
     url = app.state.config.OPENAI_API_BASE_URLS[idx]
     key = app.state.config.OPENAI_API_KEYS[idx]
+    is_o1 = payload["model"].lower().startswith("o1")
```

Comment:
I'm still hesitant to add this in, there might be other custom models with ids that start with o1 and is NOT the `o1` models from OpenAI, have you put more thought into this?

Reply:
Yes, I have considered it, but after referring to most of the models currently available on the market, none start with O1. I plan to... If there are any, I will modify the code later or change the condition to o1-xxx. However, if OpenAI releases a full version with the name O1 simply as O1, using o1-xxx might not match. So this approach has some commercial rights considerations.


The names of most models currently available on the market are as follows:
| Model                                      | Input $/1M tokens | Output$/1M tokens |
|-----------------------------------------|----------------|----------------|
| o1-preview-2024-09-12                   | $15            | $60            |
| o1-preview                                | $15            | $60            |
| o1-mini-2024-09-12                       | $3             | $12            |
| o1-mini                                   | $3             | $12            |
| gpt-4o-mini-2024-07-18                   | $0.15          | $0.6           |
| gpt-4o-mini                               | $0.15          | $0.6           |
| gpt-4o-2024-08-06                         | $2.5           | $10            |
| gpt-4o-2024-05-13                         | $5             | $15            |
| gpt-4o                                    | $5             | $15            |
| gpt-4-vision-preview                      | $10            | $30            |
| gpt-4-turbo-preview                       | $10            | $30            |
| gpt-4-turbo-2024-04-09                   | $10            | $30            |
| gpt-4-turbo                               | $10            | $30            |
| gpt-4-32k                                 | $60            | $120           |
| gpt-4-1106-preview                        | $10            | $30            |
| gpt-4-0613                                | $30            | $60            |
| gpt-4-0314                                | $30            | $60            |
| gpt-4-0125-preview                        | $10            | $30            |
| gpt-4                                     | $30            | $60            |
| gpt-3.5-turbo-instruct                    | $1.5           | $2             |
| gpt-3.5-turbo-16k-0613                    | $3             | $4             |
| gpt-3.5-turbo-16k                        | $3             | $4             |
| gpt-3.5-turbo-1106                       | $1             | $2             |
| gpt-3.5-turbo-0613                       | $1.5           | $2             |
| gpt-3.5-turbo-0125                       | $0.5           | $1.5           |
| gpt-3.5-turbo                             | $0.5           | $1.5           |
| claude-instant-1.2                       | $0.8           | $3.6           |
| claude-3-sonnet-20240229                  | $4             | $20            |
| claude-3-opus@20240229                   | $20            | $100           |
| claude-3-opus-20240229                    | $20            | $100           |
| claude-3-haiku@20240307                  | $0.4           | $2             |
| claude-3-haiku-20240307                   | $0.4           | $2             |
| claude-3-5-sonnet@20240620                | $4             | $20            |
| claude-3-5-sonnet-20240620                | $4             | $20            |
| claude-2.1                                | $11.02         | $49.59         |
| claude-2.0                                | $11.02         | $49.59         |
| chatgpt-4o-latest                        | $5             | $15            |
| Doubao-pro-128k                          | $60            | $60            |
| Gryphe/MythoMax-L2-13b                   | $0.4           | $0.4           |
| Mistral-large-2407                       | $3             | $9             |
| Qwen/Qwen2-72B-Instruct                  | $0.8           | $0.8           |
| Qwen/Qwen2-7B-Instruct                   | $0.08          | $0.08          |
| Qwen/Qwen2.5-32B-Instruct                 | $0.6           | $0.6           |
| Qwen/Qwen2.5-72B-Instruct                 | $0.8           | $0.8           |
| Qwen/Qwen2.5-7B-Instruct                  | $0.4           | $0.4           |
| THUDM/chatglm3-6b                        | $60            | $60            |
| THUDM/glm-4-9b-chat                      | $60            | $60            |
| WizardLM/WizardCoder-Python-34B-V1.0     | $0.9           | $0.9           |
| ahm-Phi-3-5-mini-instruct                 | $1             | $3             |
| ahm-Phi-3-5-vision-instruct               | $0.4           | $1.6           |
| ahm-Phi-3-medium-128k                     | $6             | $18            |
| ahm-Phi-3-medium-4k                       | $1             | $3             |
| ahm-Phi-3-small-128k                      | $1             | $3             |
| Cohere-command-r                 | $0.64          | $1.92          |
| Jamba-1-5-Large                  | $2.2           | $8.8           |
| Llama-3-1-405B-Instruct          | $5             | $15            |
| Llama-3-1-70B-Instruct           | $0.6           | $0.78          |
| Llama-3-1-8B-Instruct            | $0.3           | $0.6           |
| Llama-3-70B-Instruct             | $0.7           | $0.7           |
| Mistral-large                    | $4             | $12            |
| Mistral-large-2407               | $3             | $9             |
| command-r-plus                   | $3.84          | $19.2          |
| anthropic-3-5-sonnet-20240620             | $4             | $20            |
| babbage-002                               | $0.4           | $0.4           |
| cogview-3-plus                            | $10            | $10            |
| command                                   | $1             | $2             |
| command-light                             | $1             | $2             |
| command-light-nightly                     | $1             | $2             |
| command-nightly                           | $1             | $2             |
| command-r                                 | $0.64          | $1.92          |
| command-r-plus                            | $3.84          | $19.2          |
| dall-e-2                                  | $16            | $16            |
| dall-e-3                                  | $40            | $40            |
| davinci-002                               | $2             | $2             |
| deepseek-ai/DeepSeek-Coder-V2-Instruct   | $0.4           | $0.8           |
| deepseek-ai/DeepSeek-V2-Chat              | $0.4           | $0.8           |
| deepseek-ai/DeepSeek-V2.5                 | $0.4           | $0.8           |
| deepseek-chat                             | $0.4           | $0.8           |
| deepseek-coder                            | $0.4           | $0.8           |
| distil-whisper-large-v3                   | $5.556         | $5.556         |
| gemini-1.5-flash                          | $0.2           | $0.8           |
| gemini-1.5-flash-002                      | $0.4           | $1.6           |
| gemini-1.5-flash-8b-exp-0827              | $0.2           | $0.8           |
| gemini-1.5-flash-exp-0827                 | $0.2           | $0.8           |
| gemini-1.5-pro                            | $8             | $24            |
| gemini-1.5-pro-002                        | $8             | $24            |
| gemini-1.5-pro-exp-0801                   | $2.5           | $7.5           |
| gemini-1.5-pro-exp-0827                   | $2.5           | $7.5           |
| gemini-pro                                | $0.2           | $0.6           |
| gemma-7b-it                               | $0.1           | $0.1           |
| gemma2-9b-it                              | $0.4           | $0.4           |
| glm-3-turbo                               | $0.71          | $0.71          |
| glm-4                                     | $14.2          | $14.2          |
| glm-4-flash                               | $0.1           | $0.1           |
| glm-4-plus                                | $8             | $8             |
| glm-4v-plus                               | $2             | $2             |
| llama-3.1-405b-instruct                   | $4             | $4             |
| llama-3.1-70b                             | $0.6           | $0.6           |
| llama-3.1-70b-versatile                   | $0.6           | $0.6           |
| llama-3.1-8b-instant                      | $0.8           | $0.8           |
| llama-3.2-11b-text-preview                | $0.2           | $0.2           |
| llama-3.2-1b-preview                      | $0.2           | $0.2           |
| llama-3.2-3b-preview                      | $0.2           | $0.2           |
| llama-3.2-90b-text-preview                | $0.2           | $0.2           |
| llama3-70b-8192                           | $0.7           | $0.9 |
| llama3-8b-8192                            | $0.06          | $0.12          |
| mattshumer/Reflection-Llama-3.1-70B      | $0.2           | $0.2           |
| microsoft/WizardLM-2-8x22B                | $1.4           | $1.4           |
| mistral-large-latest                      | $4             | $12            |
| mixtral-8x7b-32768                       | $0.5           | $0.5           |
| moonshot-v1-128k                         | $8.52          | $8.52          |
| moonshot-v1-32k                          | $3.408         | $3.408         |
| moonshot-v1-8k                           | $1.704         | $1.704         |
| qwen-long                                 | $0.1           | $0.4           |
| qwen-max                                  | $7             | $21            |
| qwen-max-longcontext                      | $7             | $21            |
| qwen-plus                                 | $0.7           | $2.1           |
| qwen-turbo                                | $0.36          | $1.08          |
| qwen2.5-14b-instruct                      | $0.4           | $1.2           |
| qwen2.5-32b-instruct                      | $0.6           | $1.2           |
| qwen2.5-3b-instruct                       | $0.4           | $0.8           |
| qwen2.5-72b-instruct                      | $0.8           | $2.4           |
| qwen2.5-7b-instruct                       | $0.4           | $0.8           |
| qwen2.5-coder-1.5b-instruct               | $0.2           | $0.4           |
| qwen2.5-coder-7b-instruct                 | $0.2           | $0.4           |
| qwen2.5-math-72b-instruct                 | $0.8           | $2.4           |
| qwen2.5-math-7b-instruct                  | $0.2           | $0.4           |
| text-ada-001                              | $0.4           | $0.4           |
| text-babbage-001                          | $0.5           | $0.5           |
| text-curie-001                            | $2             | $2             |
| text-davinci-002                          | $20            | $20            |
| text-davinci-003                          | $20            | $20            |
| text-davinci-edit-001                     | $20            | $20            |
| text-embedding-3-large                    | $0.13          | $0.13          |
| text-embedding-3-small                    | $0.02          | $0.02          |
| text-embedding-ada-002                    | $0.1           | $0.1           |
| text-moderation-latest                    | $0.2           | $0.2           |
| text-moderation-stable                    | $0.2           | $0.2           |
| tts-1                                     | $15            | $15            |
| tts-1-hd                                  | $30            | $30            |
| whisper-1                                 | $100           | $100           |
| whisper-large-v3                          | $30.834        | $30.834        |
| yi-34b-chat-0205                         | $0.000355      | $0.000355      |
| yi-34b-chat-200k                         | $0.001704      | $0.001704      |
| yi-large                                  | $3             | $3             |
| yi-large-rag                              | $4             | $4             |
| yi-large-turbo                            | $1.8           | $1.8           |
| yi-medium                                 | $0.4           | $0.4           |
| yi-vl-plus                                | $0.000852      | $0.000852      |